### DISCUSSION TITLE
why choose postgresql?

### DISCUSSION THREAD

**Discussion by liukaiyuliukaiyu** [Kong/kong#12742] [Help] [answered]
Kong database has three options: postgresql, Cassandra, mysql. For kong, what is this three kinds of database application scenario, why did you choose postgresql, without having to Cassandra and mysql

**Comment by chronolaw** [comment]
A reminder: Kong 3.4 has removed the support of Cassandra.

**Comment by chronolaw** [comment, marked as answer]
This blog post may be helpful. https://konghq.com/blog/product-releases/cassandra-support-deprecated

**Reply by nowNick** [reply]
The article that @chronolaw linked answers this exact question. Here's an excerpt:
>During the years, the customers and open source community members have asked us to support other databases such as Redis, MySQL / MariaDB / Percona, Hazelcast and many others. There have been even implementations by the community to add some of those. While adding support to other databases could solve some of the problems or help Kong Gateway integrate with environments where such databases are already deployed and well-understood and approved, it is not exactly something on our focus. We want the Kong Gateway to be, first and foremost, the best possible solution as an ingress gateway (north-south). Similarly, we want the [Kuma](https://kuma.io/) and [Kong Mesh](https://konghq.com/kong-mesh) to be the best solution to deploy [service mesh](https://konghq.com/learning-center/service-mesh/what-is-a-service-mesh) (east-west). Adding support for additional databases means mostly unnecessary work and maintenance, and sometimes confusion (...)


Therefore I'll mark this response as answer.

**Comment by ADD-SP** [comment]
I haven't heard that MySQL is a backend database option of the Kong Gateway. Where did you know this?

**Reply by liukaiyuliukaiyu** [reply]
kong version 0.10.3 has the module code for mysql.lua, I also guess

**Reply by subnetmarco** [reply]
It is not officially supported, you may have found an alternative driver.

**Comment by asadali48** [comment]
Choosing PostgreSQL as your relational database management system (RDBMS) offers several advantages:

**Advanced Features:** PostgreSQL is feature-rich and supports advanced SQL features, including complex queries, window functions, common table expressions (CTEs), full-text search, and JSON/XML data types. It also supports advanced data types such as arrays, hstore, and range types.

**Extensibility:** PostgreSQL offers a robust extension framework that allows developers to extend its functionality with custom plugins and extensions. This enables the integration of additional features and capabilities tailored to specific use cases, such as geographic information systems (GIS), data warehousing, and document storage.

**Open Source and Community-driven:** PostgreSQL is open-source software, which means it is free to use, modify, and distribute. It has a vibrant and active community of developers, contributors, and users who collaborate to improve the software, fix bugs, and add new features. The community provides extensive documentation, tutorials, and support resources.

**High Performance:** PostgreSQL is known for its performance and scalability, particularly in multi-user and high-volume transactional environments. It offers advanced optimization techniques, including query optimization, indexing, and parallel query execution, to deliver high performance and throughput.

**ACID Compliance:** PostgreSQL ensures data integrity and reliability by supporting ACID (Atomicity, Consistency, Isolation, Durability) properties. It provides robust transaction management capabilities, including multi-version concurrency control (MVCC), which allows multiple transactions to access data simultaneously without interfering with each other.

**Cross-platform Compatibility:** PostgreSQL is available on multiple platforms, including Windows, Linux, macOS, and various Unix-like operating systems. This cross-platform compatibility makes it suitable for deployment in diverse environments and allows for seamless integration with existing systems and applications.

**Scalability and Replication:** PostgreSQL supports various replication methods, including streaming replication, logical replication, and built-in failover mechanisms, to achieve high availability and scalability. It also offers support for sharding and partitioning techniques to distribute data across multiple servers for horizontal scalability.

**Security:** PostgreSQL provides robust security features to protect sensitive data, including role-based access control (RBAC), SSL/TLS encryption, row-level security (RLS), and comprehensive auditing and logging capabilities. It undergoes regular security audits and updates to address potential vulnerabilities and threats.

Overall, PostgreSQL is a powerful, reliable, and versatile RDBMS that offers advanced features, performance, scalability, and security, making it an excellent choice for a wide range of applications and use cases.
